### ISSUE TITLE
Freecad up to 20220721 missing Icons in Snap

### ISSUE THREAD

**Issue by ckvsoft** [realthunder/FreeCAD#506]
Hi
![Screenshot_freecad](https://user-images.githubusercontent.com/16374447/180305719-dd329e89-7e47-4414-9adf-058ac4a5bd6a.png)

In the Freecad Stage 3 Link versions up to 2022.07.21 various icons are missing in the snap image. Could someone please fix this? See photo.
The options on the left don't work either. These are always grayed out

**Comment by ckvsoft** [comment]
![Screenshot_freecad2](https://user-images.githubusercontent.com/16374447/180306414-72baa0a9-b48d-45c9-aaa8-91ce7d2c93e4.png)

**Comment by ckvsoft** [comment]
![Screenshot_freecad3](https://user-images.githubusercontent.com/16374447/180306662-0516af80-3fff-4022-a832-c9212816b1dc.png)

**Comment by realthunder** [comment]
Did this just happen recently? Did Previous version have this problem? @ppd, any idea what might be the problem?

**Comment by ppd** [comment]
Disclaimer: I can't reproduce the issue. It smells like an icon theme is in use that does not have those icons present or that does not have a snapped version available, so some standard theme is used that also does not have those icons.

There was this change recently: https://github.com/realthunder/FreeCAD/commit/d3be8ee894f2f2d00c235069665366e212cbcd6e

> [...]so FreeCAD uses its included icons instead of the system icon theme on Linux
> Fixes missing/changed icons after installing new workbenches from
> the addon manager.

I'd try changing the parameter BaseApp/Preferences/Bitmaps/Theme/ThemeSearchPaths to `true` (or false if it is already true).

As for the unclickable sidebar items: That's much nicer in the upstream snap (see screenshot). 
![image](https://user-images.githubusercontent.com/6652223/180392272-a53590a7-08e7-45f7-ad97-daa4eb30e828.png)

It seems like nothing "clickable" gets added in the LinkStage snap. Cherry-picking whatever commit is responsible for this from upstream would really improve the UX.

**Comment by ckvsoft** [comment]
> Did this just happen recently? Did Previous version have this problem?

I had that in previous versions. Reported it now because nothing has changed up to the current version. The app images work, but I have other problems there

**Comment by ckvsoft** [comment]
> Disclaimer: I can't reproduce the issue. It smells like an icon theme is in use that does not have those icons present or that does not have a snapped version available, so some standard theme is used that also does not have those icons.
> 
> There was this change recently: [d3be8ee](https://github.com/realthunder/FreeCAD/commit/d3be8ee894f2f2d00c235069665366e212cbcd6e)
> 
> > [...]so FreeCAD uses its included icons instead of the system icon theme on Linux
> > Fixes missing/changed icons after installing new workbenches from
> > the addon manager.
> 
> I'd try changing the parameter BaseApp/Preferences/Bitmaps/Theme/ThemeSearchPaths to `true` (or false if it is already true).
> 
> As for the unclickable sidebar items: That's much nicer in the upstream snap (see screenshot).

I've tried all sorts of themes. No change. I'll try that with Theme Search Paths.

What or Where is the Upstream Snap?

**Comment by ppd** [comment]
> What or Where is the Upstream Snap?

https://snapcraft.io/freecad

However, those nice bookmarks in the sidebar of the dialog are only present on the edge channel. There's also an edge/toponaming channel for a preview of @realthunder's integration work of some of the stuff that's in this repo here.

**Comment by ckvsoft** [comment]
Ok, thought that was a different one.
I am using sudo snap install freecad --edge.

**Comment by ppd** [comment]
> I am using sudo snap install freecad --edge.

So this has the same issue for you?

**Comment by ckvsoft** [comment]
Yes, that's exactly the version I made the report about. Just thought you meant a different Snap version

**Comment by ckvsoft** [comment]
I now have this installed on my company computer. It's exactly the same there. So it should be because of the SNAP version.

I am using Kubuntu 22.04 in dark theme

**Comment by ppd** [comment]
If you'd be so kind as to report this issue  at https://github.com/freeCAD/freeCAD-snap then, that'd be nice.

I was just confused because your first third screenshot shows that you're using "FreeCAD Link Branch", i.e. the `freecad-realthunder` snap.

**Comment by ckvsoft** [comment]
Yes, exactly. I'm using the FreeCAD Link Branch version. And this one has the error. I got these with sudo snap install freecad --edge. Did I ask in the wrong git?

**Comment by ckvsoft** [comment]
Sorry
Now that got me a bit confused too.
Of course I have the version with me

sudo snap install freecad-realthunder --edge

Installed. But it's the same in both versions

**Comment by ckvsoft** [comment]
I think the problem is with Kubuntu.
I changed the KDE theme from Breeze to Fusion and it works. However, none of the other themes work.
DontUseNativeDialog also has no effect. The same dialog is always displayed.

**Comment by ppd** [comment]
> DontUseNativeDialog also has no effect. The same dialog is always displayed.

I think that only works in the freecad snap. @realthunder disabled it in freecad-realthunder permanently.

>  I changed the KDE theme from Breeze to Fusion and it works. However, none of the other themes work.

Maybe we should fix the icon theme for FreeCAD to Fusion then. I think that's possible via some env var.

**Comment by ckvsoft** [comment]
I change the Exec part in the Desktop file from
`Exec=env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/freecad-realthunder_freecad-realthunder.desktop /snap/bin/freecad-realthunder
`
to
`Exec=env QT_STYLE_OVERRIDE=Fusion  BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/freecad-realthunder_freecad-realthunder.desktop /snap/bin/freecad-realthunder
`
This works, at least until the next update

**Comment by realthunder** [comment]
You can still bring back the system file dialog by holding `Shift` while clicking the button. Note that in snap the system file dialog has trouble opening a file with external linked documents.

**Comment by ppd** [comment]
> Note that in snap the system file dialog has trouble opening a file with external linked documents.

But only if you use a host system with an old xdg desktop portal. At some point in the future, the vast majority will have a file chooser that does not proxy local files. 

@realthunder what do you think about locking the Qt theme to something that's known to work? Sounds a little drastic.

**Comment by ckvsoft** [comment]
> You can still bring back the system file dialog by holding `Shift` while clicking the button. Note that in snap the system file dialog has trouble opening a file with external linked documents.

Nice, but it works for me now and I can live well with the fusion theme

**Comment by ppd** [comment]
@ckvsoft you can always copy the desktop file to your home directory. In this way, the changes will never be overwritten by updates.

**Comment by ckvsoft** [comment]
@ppd  Yes that is actually true, thx

**Comment by ckvsoft** [comment]
Of course, this only works if I open it via the desktop link. When I say open file with... not. I still don't have a solution for this. Of course, if I set the theme to Fusion, it already works.
